### ISSUE TITLE
Doesn't install local package's dependencies

### ISSUE THREAD

**Issue by Connum** [body-builder/install-local-dependencies#2]
Maybe I'm missing the point, but using this does not install any local packages' dependencies, so it's almost the same as simply using
```
"dependencies": {
  "local-package": "file:../local-packege"
}
```
Is there a way to have the dependencies installed to the dependent package's `node_modules`?

**Comment by bencergazda** [comment]
It should install `local-package`, its `dependencies` and `peerDependencies` (but not `devDependencies`) to your project's node_modules. 

Did you run `install-local-dependencies` from the project root?

**Comment by thebrownfox** [comment]
Had same issue. Seems like this works.

https://github.com/npm/cli/issues/2339#issuecomment-1111228605

**Comment by bencergazda** [comment]
I checked it now. It indeed did not work **for first run** with **npm@8.15.0**, it produced the same result as running `$npm install`.

I started do find the problem, decreased npm version to **8.0.0**, but since then, I cannot reproduce the issue even with **8.15.0** or **8.19.3**. Further investigation would be required, I can do it, in case there's a need for it.

It still works ok with Yarn and PNPM.

**Comment by bencergazda** [comment]
I think I have found the problem. There might be cases when we restore the project's package.json too early, before npm reads it in its install task. And because of this, instead of installing the created tarball, npm simply links the package, as we hadn't used `install-local-dependencies` at all.

@Connum @thefoxie I am going to publish a new release with the fix in the next couple days, hopefully this will solve the issue you reported.

The `--install-links` flag might be a good solution indeed, in case you are not in a workspace environment, and/or you don't need the watcher (`watch-local-dependencies`) functionality. Btw, there's a good chance that the watcher works also in an `--install-links` installation, but I haven't tested it yet.

**Comment by bencergazda** [comment]
Published the fix in [v0.2.5](https://github.com/body-builder/install-local-dependencies/releases/tag/v0.2.5). Thanks for the report. Feel free to reopen this issue if something is still not OK.
